go/service/gregor.go

```diff
@@ -96,6 +146,10 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
 
+	// Send message to local state machine
+	g.gregorCli.ConsumeMessage(m)
```

Comment:
Maybe that's a bad idea though: https://github.com/golang/net/blob/master/context/pre_go17.go#L278-L279